app/src/main/java/io/gnosis/safe/ui/safe/settings/safe/AdvancedSafeSettingsFragment.kt

```diff
@@ -0,0 +1,130 @@
+package io.gnosis.safe.ui.safe.settings.safe
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.LinearLayout
+import androidx.annotation.StringRes
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.Observer
+import androidx.navigation.fragment.findNavController
+import io.gnosis.data.models.SafeInfo
+import io.gnosis.safe.R
+import io.gnosis.safe.ScreenId
+import io.gnosis.safe.databinding.FragmentAdvancedSafeSettingsBinding
+import io.gnosis.safe.di.components.ViewComponent
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.base.BaseViewBindingFragment
+import io.gnosis.safe.ui.safe.settings.view.LabeledAddressItem
+import io.gnosis.safe.ui.safe.settings.view.SettingItem
+import io.gnosis.safe.utils.dpToPx
+import pm.gnosis.model.Solidity
+import pm.gnosis.svalinn.common.utils.snackbar
+import pm.gnosis.svalinn.common.utils.visible
+import timber.log.Timber
+import java.math.BigInteger
+import javax.inject.Inject
+
+class AdvancedSafeSettingsFragment : BaseViewBindingFragment<FragmentAdvancedSafeSettingsBinding>() {
+
+    @Inject
+    lateinit var viewModel: AdvancedSafeSettingsViewModel
+
+    override fun screenId(): ScreenId? = ScreenId.SETTINGS_SAFE_ADVANCED
+
+    override fun inflateBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentAdvancedSafeSettingsBinding =
+        FragmentAdvancedSafeSettingsBinding.inflate(inflater, container, false)
+
+    override fun inject(component: ViewComponent) {
+        component.inject(this)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setHasOptionsMenu(true)
+        (activity as AppCompatActivity).setSupportActionBar(binding.advancedAppSettingsToolbar)
+        (activity as AppCompatActivity).supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        viewModel.state.observe(viewLifecycleOwner, Observer { state ->
+            when (val viewAction = state.viewAction) {
+                is LoadSafeInfo -> setUi(state.isLoading, viewAction.safeInfo)
+                is BaseStateViewModel.ViewAction.ShowError -> handleError(viewAction.error)
+                else -> setUi(state.isLoading)
+            }
+        })
+        viewModel.load()
+    }
+
+    private fun setUi(isLoading: Boolean, safeInfo: SafeInfo? = null) {
+        toggleLoading(isLoading)
+        safeInfo?.let { setSafeInfo(it) }
+    }
+
+    private fun setSafeInfo(safeInfo: SafeInfo) {
+        with(binding) {
+            fallbackHandlerContainer.removeAllViews()
```

Comment:
there can be only one fallback handler or none at all for the older contract versions, right?

Reply:
This is needed because, if the Fallback handler is not set the view should be a `SettingsItem` which is 60dp high, instead of having the label/blockies/addres/link icon combined view. I am not too fond either of this :/

Reply:
I think there could be a situation where the fallback handler could be set to an address that isn't the default. Safes with master copy version lower than 1.1.1  have the fallback handler set to null.

Reply:
![image](https://user-images.githubusercontent.com/17750984/87278936-098d5180-c4e6-11ea-8bc5-47bc994f5e40.png)
so its 68dp vs 60dp when it is not set. And it makes more effort to handle this case in the fallback handler view vs using settings view?